### PR TITLE
chore(api): Switch to using serializers over TypedDict for response

### PR DESCRIPTION
**Pull request by schew2381**
Per https://develop.sentry.dev/api/public/#success-responses:

> Note that we HIGHLY recommend using the first method when your endpoint returns a single object and utilizes an existing serializer. The first method ensures the documentation stays up-to-date as the endpoint evolves, and provides a higher quality API experience for our users.

### CODE REVIEW

File: src/sentry/monitors/endpoints/organization_monitor_checkin_index.py [hunk 1]
```diff
@@ -25,7 +26,9 @@
 @region_silo_endpoint
 @extend_schema(tags=["Crons"])
 class OrganizationMonitorCheckInIndexEndpoint(MonitorEndpoint):
-    public = {"GET"}
+    publish_status = {
+        "GET": ApiPublishStatus.PUBLIC,
+    }
```
Comment (schew2381) [inline src/sentry/monitors/endpoints/organization_monitor_checkin_index.py:31]: For some reason the earlier API code ownership change didn't catch this, so manually fixing it here

Comment (JoshFerge) [reply src/sentry/monitors/endpoints/organization_monitor_checkin_index.py:31]: might be worth seeing if any other endpoints are like this

Comment (schew2381) [reply src/sentry/monitors/endpoints/organization_monitor_checkin_index.py:31]: I did a quick check by ctrl-f `public = {` and this was the only one that popped up

File: src/sentry/monitors/serializers.py [hunk 2]
```diff
@@ -115,16 +131,19 @@ def _expand(self, key) -> bool:
         return key in self.expand
 
 
-class MonitorSerializerResponse(TypedDict):
+class MonitorCheckInSerializerResponseOptional(TypedDict, total=False):
+    groups: List[str]
```
Comment (schew2381) [inline src/sentry/monitors/serializers.py:135]: This was originally `group_id`, but it was changed to `groups` in the serializer response without being changed in the TypedDict.

**Review by nhsiehgit — APPROVED** [review]

**Pull request merged**